sonic-xcvrd/xcvrd/xcvrd.py

```diff
@@ -1376,6 +1381,11 @@ def task_worker(self):
                     # Skip if these essential routines are not available
```

Comment:
It will be good to understand when the Attribute error exception happens. Ideally the below line "except Exception as e:" is a superset and should cover all exceptions.

Also what is the exact error/exception when the eeprom data is corrupted/bad -- sue to bad optics

Reply:
@judyjoseph Yes - the below line "except Exception as e:" covers all exceptions apart from AttributeError. However, for AttributeError, the CMIS SM is currently being transitioned to `CMIS_STATE_READY` and not `CMIS_STATE_FAILED`.

In case of bad optics, I have seen KeyError and TypeError so far.